include/boost/geometry/extensions/triangulation/strategies/cartesian/detail/precise_math.hpp

```diff
@@ -376,6 +314,47 @@ inline RealNumber orient2d(std::array<RealNumber, 2> const& p1,
     return(D[D_nz - 1]);
 }
 
+// see page 38, Figure 21 for the calculations, notation follows the notation in the figure.
+template
+<
+    typename RealNumber,
+    std::size_t Robustness = 3
+>
+inline RealNumber orient2d(std::array<RealNumber, 2> const& p1,
+                           std::array<RealNumber, 2> const& p2,
+                           std::array<RealNumber, 2> const& p3)
+{
+    if(Robustness == 0) {
+        return (p1[0]-p3[0])*(p2[1]-p3[1]) - (p1[1]-p3[1])*(p2[0] - p3[0]);
```

Comment:
Can you add spaces here between all mathematical operators? `(p1[0] - p3[0]) * `